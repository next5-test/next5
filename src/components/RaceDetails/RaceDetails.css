.race {
  margin-top: 10px;
}

.race .header {
  border-radius: 6px 6px 0 0;
  -moz-border-radius: 6px 6px 0 0;
  -webkit-border-radius: 6px 6px 0 0;
  background: #252525;
  line-height: 40px;
  text-transform: uppercase;
  font-weight: 400;
  color: #fff;
  padding-left: 10px;
}
  
.race .content {
  background: #f0efee;
  border-radius: 0 0 6px 6px;
  -moz-border-radius: 0 0 6px 6px;
  -webkit-border-radius: 0 0 6px 6px;
  border-bottom: 1px solid #e2e2e2;
  padding: 5px;
}
  
.race table {
  width: 100%;
}

.race td {
  background-color: #f2f1f1;
  color: #252525;
  border-bottom: 1px solid #e8e7e7;
  border-top: 1px solid #fff;
  line-height: 35px;
}

.race tr:nth-child(odd) td {
  background: #ECECEC;
}

.race tr:hover td {
  background-color: #fafafa;
  -webkit-transition: background-color 200ms linear;
  -moz-transition: background-color 200ms linear;
  -o-transition: background-color 200ms linear;
  -ms-transition: background-color 200ms linear;
  transition: background-color 200ms linear;
}

.position{
  text-align: center;
}
  