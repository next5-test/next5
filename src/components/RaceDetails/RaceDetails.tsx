import * as React from 'react';
import './RaceDetails.css';
import { CompetitorType } from '../../types';
import { Link } from 'react-router-dom';

interface RaceDetailsProps {
  competitors: CompetitorType[];
  name: string;
}

const RaceDetails = ({ competitors, name }: RaceDetailsProps) => (
  <div className="race">
    <Link to="/">Back</Link>
    <div className="header">
      {name}
    </div>
    <div className="content">
      <table cellSpacing="0" cellPadding="0">
        <tbody>
          {competitors.map((competitor: CompetitorType) => (
            <tr key={competitor.Name} className="horses" >
              <td className="position">{competitor.PositionNumber}</td>
              <td>{competitor.Name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  </div>
);

export default RaceDetails;