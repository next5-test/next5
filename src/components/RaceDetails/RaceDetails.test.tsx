import * as React from 'react';
import * as renderer from 'react-test-renderer';
import RaceDetails from './RaceDetails';
import {
    BrowserRouter as Router
  } from 'react-router-dom';
  
test('<RaceDetails> renders OK', () => {
  const component = renderer.create(<Router><RaceDetails competitors={[]} name="name" /></Router>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});