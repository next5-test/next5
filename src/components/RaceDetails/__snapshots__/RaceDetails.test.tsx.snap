// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RaceDetails> renders OK 1`] = `
<div
  className="race"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Back
  </a>
  <div
    className="header"
  >
    name
  </div>
  <div
    className="content"
  >
    <table
      cellPadding="0"
      cellSpacing="0"
    >
      <tbody />
    </table>
  </div>
</div>
`;
